From: Tony Camuso <tcamuso@redhat.com>
Date: Thu, 21 Oct 2010 11:23:00 -0400
Subject: [pci] include DL580 G7 in bfsort whitelist
Message-id: <20101021112300.11545.84949.sendpatchset@localhost6.localdomain6>
Patchwork-id: 28890
O-Subject: RHEL5.6 [PATCH 1/1] Include DL580 G7 in bfsort whitelist
Bugzilla: 644879
RH-Acked-by: Peter Martuccelli <peterm@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>

commit 5b0e76a68bc06a8771f953e0383226e9675f5e09
Author: Tony Camuso <camuso@camuso.(none)>
Date:   Wed Oct 20 10:26:38 2010 -0400

    Description
    ===========

    The DL580 G5 is in the whitelist, but the G7 is not. When
    Amazon upgraded their DL580 G5 to G7, device names changed,
    and their scripts consequently failed.

    The G5 was originally added to the bfsort whitelist, because
    device naming of embedded PCI devices changed when PCI option
    cards were installed.

    This was a problem for RHEL customers having install scripts
    for large numbers of systems. Customizing install scripts
    for variations in platform device naming would be problematic
    in a data center having hundreds or thousands of systems.

    There are other platforms making the bfsort callback for the
    very same reasons.

    This patch must also be included as a hotfix for RHEL5.4 for
    Amazon and carried forward to future RHEL5 point releases.

    Bugzilla
    ========
    Bug 644879 - RHEL5.6 Include DL580 G7 in bfsort whitelist
    https://bugzilla.redhat.com/show_bug.cgi?id=644879

    Upstream Status
    ===============

    The original DL580 G5 patch was rejected upstream, because it
    would have made device naming different from one generation of
    DL580 to the next. When the patch was introduced to RHEL5.4,
    HP issued a Customer Advisory concerning the different device
    naming behavior when upgrading from earlier generations of
    DL580.

    Testing
    =======

    This patch was tested for functionality and regressions. A new
    BIOS is required, since the current BIOS uses the incorrect
    value for the dmi vendor ID field. A side-built BIOS was
    provided for testing, and the officially released BIOS is
    scheduled for Nov 2010.

    Stat
    ====
     arch/i386/pci/common.c |    8 ++++++++
     1 files changed, 8 insertions(+), 0 deletions(-)

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/i386/pci/common.c b/arch/i386/pci/common.c
index 9004085..88a2a4a 100644
--- a/arch/i386/pci/common.c
+++ b/arch/i386/pci/common.c
@@ -354,6 +354,14 @@ static struct dmi_system_id __devinitdata pciprobe_dmi_table[] = {
 	},
 	{
 		.callback = set_bf_sort,
+		.ident = "HP ProLiant DL580 G7",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant DL580 G7"),
+		},
+	},
+	{
+		.callback = set_bf_sort,
 		.ident = "HP ProLiant xw460c",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "HP"),
